bpf/lib/nodeport.h

```diff
@@ -1368,20 +1435,22 @@ int tail_nodeport_ipv4_dsr(struct __ctx_buff *ctx)
 		goto drop_err;
 	}
 
+	addr = ctx_load_meta(ctx, CB_ADDR_V4);
+	port = (__u16)ctx_load_meta(ctx, CB_PORT);
 #if DSR_ENCAP_MODE == DSR_ENCAP_IPIP
 	ret = dsr_set_ipip4(ctx, ip4,
-			    ctx_load_meta(ctx, CB_ADDR_V4),
+			    addr,
 			    ctx_load_meta(ctx, CB_HINT), &ohead);
 #elif DSR_ENCAP_MODE == DSR_ENCAP_NONE
 	ret = dsr_set_opt4(ctx, ip4,
-			   ctx_load_meta(ctx, CB_ADDR_V4),
-			   ctx_load_meta(ctx, CB_PORT), &ohead);
+			   addr,
+			   port, &ohead);
```

Comment:
k8s-1.24-kernel-net-next:
This, or other changes around it, are likely the cause for the failure:

```
[...]
1887224-; if (dsr_is_too_big(ctx, tot_len)) {                          
1887225-106: (26) if w1 > 0x5dc goto pc+65    ; R1=scalar(umax=1500,var_off=(0x0; 0x7ff)
[...]
1887331-171: (05) goto pc+353                                          
1887332-; if (ctx_store_bytes(ctx, ETH_HLEN + sizeof(*ip4)
[...]
1887777-from 106 to 172: R0=map_value(off=0,ks=4,vs=28,imm=0) R1=scalar(umin=1501,umax=65535,var_off=(0x0; 0xffff)) R2=pkt_end(off=0,imm=0) R3=scalar(umax=4294967295,var_off=(0x0; 0xffffffff)) R4=P0 R5=pkt(off=32,r=34,imm=0) R6=ctx(off=0,imm=0) R7=scalar(umax=4294967295,var_off=(0x0; 0xffffffff)) R8=scalar(umax=65535,var_off=(0x0; 0xffff)) R9=pkt(off=0,r=34,imm=0) R10=fp0 fp-96=????mmmm fp-160=????0000 fp-168=00000000 fp-176=00000000 fp-184=00000000 fp-192=00000000 fp-200=00000000 fp-208=00000000 fp-216= fp-224=2
1887778-; return (void *)(unsigned long)ctx->data;
1887779-172: (61) r1 = *(u32 *)(r6 +0)        ; R1_w=pkt(off=0,r=0,imm=0) R6=ctx(off=0,imm=0)
1887780-; return (void *)(unsigned long)ctx->data_end;
1887781-173: (63) *(u32 *)(r10 -232) = r1
1887782-invalid size of register spill
1887783-processed 374 insns (limit 1000000) max_states_per_insn 1 total_states 27 peak_states 27 mark_read 10
1887784-
1887785-libbpf: -- END LOG --
1887786-libbpf: failed to load program 'tail_nodeport_ipv4_dsr'